lib/Sema/CSDiagnostics.cpp

```diff
@@ -2352,3 +2352,133 @@ bool InaccessibleMemberFailure::diagnoseAsError() {
   emitDiagnostic(Member, diag::decl_declared_here, Member->getFullName());
   return true;
 }
+
+bool ImplicitCoercionToAnyFailure::diagnoseAsError() {
+  auto E = getAnchor();
+  auto subExpr = E->getValueProvidingExpr();
+  auto &TC = getTypeChecker();
+  Decl *decl = nullptr;
+
+  if (auto *UDE = dyn_cast<UnresolvedDotExpr>(E)) {
+    auto name = UDE->getName();
+    auto parent = getDC()->getParentForLookup();
+    SmallVector<ValueDecl *, 4> lookupResults;
+    parent->lookupQualified(FromType->lookThroughAllOptionalTypes(), name,
```

Comment:
This should be retrieved from constraint system. It holds the choice associated with given member reference, locator would be `(E, ConstraintLocator::Member)` or `(E, ConstraintLocator::ConstructorMember)` depending on that the base was.

Reply:
Done!